src/tv2-common/actions/CoreActionExecutionContext.ts

```diff
@@ -0,0 +1,238 @@
+	IBlueprintPart,
+	IBlueprintPartInstance,
+	IBlueprintPiece,
+	IBlueprintPieceDB,
+	IBlueprintPieceInstance,
+	IBlueprintResolvedPieceInstance,
+	PackageInfo,
+	Time
+} from 'blueprints-integration'
+import { literal, PartMetaData, PieceMetaData } from 'tv2-common'
+import { ITV2ActionExecutionContext } from './context'
+
+export class CoreActionExecutionContext implements ITV2ActionExecutionContext {
+	public studioId: string
+	public isTV2Context: true = true
+
+	private modifiedParts: Set<'current' | 'next'> = new Set()
+
+	constructor(readonly core: IActionExecutionContext) {
+		this.studioId = core.studioId
+	}
+
+	public async getPartInstance(part: 'current' | 'next'): Promise<IBlueprintPartInstance<unknown> | undefined> {
+		return this.core.getPartInstance(part)
+	}
+
+	public async getPieceInstances(part: 'current' | 'next'): Promise<Array<IBlueprintPieceInstance<PieceMetaData>>> {
+		return this.core.getPieceInstances(part) as Promise<Array<IBlueprintPieceInstance<PieceMetaData>>>
+	}
+
+	public async getResolvedPieceInstances(
+		part: 'current' | 'next'
+	): Promise<Array<IBlueprintResolvedPieceInstance<PieceMetaData>>> {
+		return this.core.getResolvedPieceInstances(part) as Promise<Array<IBlueprintResolvedPieceInstance<PieceMetaData>>>
+	}
+
+	public async findLastPieceOnLayer(
+		sourceLayerId: string | string[],
+		options?: {
+			excludeCurrentPart?: boolean
+			originalOnly?: boolean
+			pieceMetaDataFilter?: any
+		}
+	): Promise<IBlueprintPieceInstance<PieceMetaData> | undefined> {
+		return this.core.findLastPieceOnLayer(sourceLayerId, options) as Promise<
+			IBlueprintPieceInstance<PieceMetaData> | undefined
+		>
+	}
+
+	public async findLastScriptedPieceOnLayer(
+		sourceLayerId: string | string[],
+		options?: { excludeCurrentPart?: boolean; pieceMetaDataFilter?: any }
+	): Promise<IBlueprintPiece<PieceMetaData> | undefined> {
+		return this.core.findLastScriptedPieceOnLayer(sourceLayerId, options) as Promise<
+			IBlueprintPiece<PieceMetaData> | undefined
+		>
+	}
+
+	public async getPartInstanceForPreviousPiece(
+		piece: IBlueprintPieceInstance<PieceMetaData>
+	): Promise<IBlueprintPartInstance<unknown>> {
+		return this.core.getPartInstanceForPreviousPiece(piece)
+	}
+
+	public async getPartForPreviousPiece(
+		piece: IBlueprintPieceDB<PieceMetaData>
+	): Promise<IBlueprintPart<unknown> | undefined> {
+		return this.core.getPartForPreviousPiece(piece)
+	}
+
+	public async stopPiecesOnLayers(sourceLayerIds: string[], timeOffset?: number): Promise<string[]> {
+		return this.core.stopPiecesOnLayers(sourceLayerIds, timeOffset)
+	}
+
+	public async stopPieceInstances(pieceInstanceIds: string[], timeOffset?: number): Promise<string[]> {
+		return this.core.stopPieceInstances(pieceInstanceIds, timeOffset)
+	}
+
+	public async takeAfterExecuteAction(take: boolean): Promise<boolean> {
+		return this.core.takeAfterExecuteAction(take)
+	}
+
+	public notifyUserError(message: string, params?: { [key: string]: any }): void {
```

Comment:
Unsure if this is a complete copy of an older file.

If it returns `void`, then it is confusing if it has a return statement, with a possible outcome.
When compiled down to JS, it might on runtime actually return something.

If a value is returned on runtime, should it continue onward, even thought calls to this method, would indicate that nothing is returned?

Same comment for `notifyUserWarning(...)` below.

Reply:
This is a decorator class. If it didn't exist because we didn't want the added behavior, the method from the original class could also return a value in the case that you're describing, meaning at runtime our class actually behaves as close as possible to the decorated class, which to me seems like a good thing.